### PR TITLE
added extra option -d

### PR DESCRIPTION
**Pull request by alexzk1**
Well, sometimes I want screenshot fast-running things, like flying inside game or so. I don't have time to click all that boxes and/or it may crash game.
So this one just addsoption -d <folder> where program just dumps files with robust namings.
Also some needed changes to avoid copy-paste.

### CODE REVIEW

**Comment by tsujan** [comment]
Thanks for your contribution! Please:

(1) Clean up the PR (`CMakeLists.txt.user`?!).

(2) Respect the code style of the original author (number of spaces, no space between `if` and `{`, etc. -- a big part of the diff is about that).

**Comment by alexzk1** [comment]
well, yeh, I changed code styling on late commits ..however I still guessing options for it, can you provide styling options for "Artistic Style" ? 
Also different files has different settings.

**Comment by tsujan** [comment]
@alexzk1 Thanks! It's much better now.

Please be patient as its review might take a while -- unless another dev already has the required time.

**Comment by tsujan** [comment]
One more thing: This part of lximage-qt should be completely rewritten, based on KWindowSystem. We should remove X11 codes as far as possible. But that's a job for future.

File: src/screenshotdialog.cpp [hunk 1]
```diff
@@ -183,9 +210,84 @@ void ScreenshotDialog::doScreenshot() {
...
 
   deleteLater(); // destroy ourself
 }
+
+static QString buildNumericFnPart() {
+  //we may have many copies running with no gui, for example user presses hot keys fast
+  //so they must have different file names to save, lets do it time + pid
+  const auto now = QDateTime::currentDateTime().toMSecsSinceEpoch();
+  const auto pid = getpid();
+  return QString(QString::fromLatin1("%1_%2")).arg(now).arg(pid);
```
Comment (tsujan) [inline src/screenshotdialog.cpp]: Please change this to:

```c++
return QStringLiteral("%1_%2").arg(now).arg(pid);
```

File: src/screenshotdialog.cpp [hunk 2]
```diff
@@ -183,9 +210,84 @@ void ScreenshotDialog::doScreenshot() {
...
+  return (result.isEmpty())?QString::fromUtf8("UKNOWN"):result;
+}
+
+void ScreenshotDialog::cmdTopShotToDir(const QString &path) {
+
+  WId activeWid = activeWindowId();
+  const QRect rect = (activeWid) ? windowFrame(activeWid) : QRect{0, 0, -1, -1};
+  QImage img{takeScreenshot(QApplication::desktop()->winId(), rect, false)};
+
+  const QString filename = path % QString::fromLatin1("/") % getWindowName(activeWid) % QString::fromUtf8("_") % buildNumericFnPart();
```
Comment (tsujan) [inline src/screenshotdialog.cpp]: Please change all instances of `QString::fromLatin1()` to `QStringLiteral()`.

File: src/screenshotdialog.cpp [hunk 3]
```diff
@@ -183,9 +210,84 @@ void ScreenshotDialog::doScreenshot() {
...
+            }
+            result = QString::fromStdString(ss.str());
+            break;
+          }
+        }
+
+      }
+    }
+  }
+  return (result.isEmpty())?QString::fromUtf8("UKNOWN"):result;
```
Comment (tsujan) [inline src/screenshotdialog.cpp]: Please change all instances of `QString::fromUtf8()` to `QStringLiteral()`.

**Comment by alexzk1** [comment]
Done. Changed string join too.
However can't compile now because reverted system libfm-qt to older. Older does not compile, newer which compiles, leads to crashes of file manager. 
So try yourself too to see if I didn't do typos.

**Comment by tsujan** [comment]
> newer which compiles, leads to crashes of file manager.

LXQt components should be upgraded together. For example, when libfm-qt is upgraded, pcmanfm-qt, lximage-qt and lxqt-qrchiver should be too.

> So try yourself too to see if I didn't do typos.

I will. The code is good, in general.

Why did you add PID to file name? Wasn't the current date in milliseconds enough?

**Comment by alexzk1** [comment]
>Why did you add PID to file name? Wasn't the current date in milliseconds enough?

Because u can launch couple copies very fast (on many cores it is easy) and all them may have same ms stamp. So PID ensures name is unique, and still I do extra check there and it may add extra sequental number. For example that may happen if between 2 screenshots network time correction happened.

**Comment by tsujan** [comment]
OK. I guess you've considered all possible scenarios.

**Comment by alexzk1** [comment]
>OK. I guess you've considered all possible scenarios.
Not really, but I feel that probability to overwrite is very low.
To catch really ALL need atomic write :D That will require to have some running daemon which manages names...

**Comment by tsujan** [comment]
I see no problem. GTM.

Would you please squash all of your commits, so that I can merge it?

**Comment by tsujan** [comment]
Never mind! GitHub had a button for that :)

**Comment by tsujan** [comment]
Oh, GitHub's squash-and-merge button wasn't good :( I shouldn't use it anymore.

**Comment by alexzk1** [comment]
Offtopic:
That's why I needed that -d option, to do that: 
image

**Comment by tsujan** [comment]
> That's why I needed that -d option, to do that:

I knew. It was a good addition.

However, we really need to get rid of X11 codes in lximage-qt, as far as possible.

**Pull request merged**